### PR TITLE
Upgrade dependency: ganache-core@2.9.1

### PR DESCRIPTION
**Pull request by CruzMolina**
* @truffle/artifactor: 2.8.0 →  2.9.1
* @truffle/contract: 2.8.0 →  2.9.1
* @truffle/core: 2.8.0 →  2.9.1
* @truffle/debugger: 2.8.0 →  2.9.1
* @truffle/deployer: 2.8.0 →  2.9.1
* @truffle/environment: 2.8.0 →  2.9.1
* @truffle/hdwallet-provider: 2.8.0 →  2.9.1
* @truffle/interface-adapter: 2.8.0 →  2.9.1
* @truffle/provider: 2.8.0 →  2.9.1
* truffle: 2.8.0 →  2.9.1

### CODE REVIEW

File: packages/deployer/test/errors.js [hunk 1]
```diff
@@ -213,8 +213,8 @@ describe("Error cases", function() {
       assert.fail();
     } catch (err) {
       assert(err.message.includes("Loops"));
-      assert(err.message.includes("code couldn't be stored"));
-      assert(err.message.includes("check your gas limit"));
+      assert(err.message.includes("ran out of gas"));
+      assert(err.message.includes("network config or deployment parameters"));
```
Comment (haltman-at) [inline packages/deployer/test/errors.js]: I assume this is just changes in error messages?

File: packages/contract/test/deploy.js [hunk 2]
```diff
@@ -225,8 +224,8 @@ describe("Deployments", function() {
 
       assert(multiplier === 1.25, "Multiplier should be initialized to 1.25");
       assert(
-        multiplier * estimate > block.gasLimit,
-        "Multiplied estimate should be too high"
+        multiplier * estimate < block.gasLimit,
```
Comment (haltman-at) [inline packages/contract/test/deploy.js]: Huh, what's the reason for this change?

File: packages/deployer/test/errors.js (hunk 1 above)
Comment (CruzMolina) [reply packages/deployer/test/errors.js]: there was a regression that's been fixed in this release. I'll change this commit to use the original (now fixed in this release) output we were looking for https://github.com/trufflesuite/truffle/blame/be86b8eed120af9dca7d2ea1ee751a54ded200fc/packages/deployer/test/errors.js#L215-L218.

Comment (haltman-at) [reply packages/deployer/test/errors.js]: Oh yeah that whole thing... cool.

File: packages/contract/test/deploy.js (hunk 2 above)
Comment (CruzMolina) [reply packages/contract/test/deploy.js]: This test is/was confusing because the `it` statement says the multiplied estimate *should* be less than the `blockLimit` and we were previously asserting that it *was* higher than the `blockLimit`.

After chatting with @nicholasjpaterno about this test case, it appears the gas cost for creating the `Example` contract got much cheaper (by 2/3rds) due to a dirty storage slot bug fix in `istanbul` (which ganache now defaults to). See [EIP 2200](https://eips.ethereum.org/EIPS/eip-2200) & [EIP 2028](https://eips.ethereum.org/EIPS/eip-2028).

Thus, it made sense to reverse the sign since this appears to be in line with the original intent of this specific test.

**Comment by CruzMolina** [comment]
no wait!

**Comment by CruzMolina** [comment]
need to pass the `hardFork` option, since the default changed. been pairing with @nicholasjpaterno, there be bugs in the `istanbul` logic.
